### PR TITLE
[pybind11] New recipe

### PR DESCRIPTION
**Pull request by pabloferz**

### CODE REVIEW

**Comment by giordano** [comment]
Working only on Windows? :joy:

**Comment by pabloferz** [comment]
Somehow cmake fails to detect `distutils` on Azure but, it works locally on a build environment without `apk add py3-distutils-extra` which of course we don't need since we are only copying headers

File: P/pybind11/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DPYTHON_EXECUTABLE="${bindir}/python3.8${exeext}" \
+    -DPYBIND11_TEST=OFF
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [AnyPlatform()]
```
Comment (giordano) [inline P/pybind11/build_tarballs.jl:30]: @pabloferz as far as I can tell this builder produces only header files, and from what I can see they're platform-independent (at least between the 3 platforms I was able to check it).  In this case we can build for [`AnyPlatform`](https://juliapackaging.github.io/BinaryBuilder.jl/dev/reference/#BinaryBuilder.AnyPlatform).  If you're happy with these change we can merge it

File: P/pybind11/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# No products: pybind11 is a pure header library
+products = Product[
```
Comment (giordano) [inline P/pybind11/build_tarballs.jl:33]: Do you want to specify any product or just need this empty?

Comment (pabloferz) [reply P/pybind11/build_tarballs.jl:33]: Usually one only needs the header to use them with another project, so empty is fine.

**Comment by PallHaraldsson** [comment]
Where is pybind11 used (and how, I just stumbled on this by accident)? I'm not sure, if this is done to resurrect this package (for e.g. calling PHP; it uses pybind):

https://gitlab.com/wavexx/Polyglot.jl

**Comment by pabloferz** [comment]
I know nothing about Polyglot.jl so I can't really comment on their use case. That said, there is a bunch of C++ codes with python APIs that expose C++ through pybind11, actually https://github.com/JuliaInterop/CxxWrap.jl is Julia's pybind11 equivalent.

Sometimes pybind is cleanly used outside the C++ APIs in which case you can almost pretty easily translate pybind11 code to CxxWrap, but sometimes C++ and pybind11 code gets somewhat entangled that I thought on getting here for cases where you want to use CxxWrap, but cannot get pybind11 out of the way.

**Comment by PallHaraldsson** [comment]
Sorry for the noise, I thought I remembered that package using pybind, but I see it (directly) uses only python-bond (I'll look into those package later, to see if related, or uses there other one, as I wanted to get that package working again):

https://www.thregr.org/~wavexx/software/python-bond/#practical-examples

**Pull request merged**